tests/test_auth.py

```diff
@@ -0,0 +1,196 @@
+        jwt.utils.base64url_encode(b'{"not valid'),
+        jwt.utils.base64url_encode(b"some bytes"),
+    )).decode()
+    assert jwt_manager._decode_jwt(encoded) is None
+
+
+def test_get_auth_cookie(jwt_manager):
+    jwt_manager.cookie_name = 'auth-cookie'
+
+    headers = {'Cookie': 'auth-cookie=foo'}
+    assert jwt_manager._get_auth_cookie(headers).value == 'foo'
+    headers = {'Cookie': 'auth-cookie=foo; not-auth-cookie=bar'}
+    assert jwt_manager._get_auth_cookie(headers).value == 'foo'
+    headers = {'Cookie': 'not-auth-cookie=foo'}
+    assert jwt_manager._get_auth_cookie(headers) is None
+
+
+def test_format_date_for_cookie_header():
+    assert JwtManager._format_date_for_cookie_header(0) == "Thu, 01 Jan 1970 00:00:00 GMT"
+
+    assert JwtManager._format_date_for_cookie_header(1) == "Thu, 01 Jan 1970 00:00:01 GMT"
+
+    assert JwtManager._format_date_for_cookie_header(1_000_000) == "Mon, 12 Jan 1970 13:46:40 GMT"
+
+    assert JwtManager._format_date_for_cookie_header(1_000_000_000) == "Sun, 09 Sep 2001 01:46:40 GMT"
+
+
+@mock.patch(f'{MODULE}.time')
+@mock.patch(f'{MODULE}.urllib.request')
+def test_retrieve_blacklist(mock_request, mock_time, jwt_manager):
+    mock_request.urlopen(jwt_manager.black_list_endpoint).read.return_value = b'''{"blacklist": {"foo": "bar"}}'''
+    mock_time.return_value = 123
+    jwt_manager._retrieve_blacklist()
+    assert jwt_manager.black_list == {'foo': 'bar'}
+    assert jwt_manager.black_list_last_retrieved == 123
+
+
+@mock.patch(f'{MODULE}.JwtManager._retrieve_blacklist')
+def test_in_blacklist(mock_retrieve_blacklist, jwt_manager):
+    jwt_manager.black_list = {
+        'blacklisted_user': 100,
+    }
+
+    profile = UserProfile(
+        user_id='blacklisted_user',
+        token='test_token',
+        groups=['test_group1', 'test_group2'],
+        first_name='test_first_name',
+        last_name='test_last_name',
+        email='test_email',
+        iat=75
+    )
+    assert jwt_manager._in_blacklist(profile) is True
+    profile.iat = 115
+    assert jwt_manager._in_blacklist(profile) is False
+
+
+@mock.patch(f'{MODULE}.time', autospec=True)
+def test_jwt_payload_from_user_profile(mock_time, jwt_manager):
+    profile = UserProfile(
+        user_id='test_user_id',
+        token='test_token',
+        groups=['test_group1', 'test_group2'],
+        first_name='test_first_name',
+        last_name='test_last_name',
+        email='test_email',
+    )
+    mock_time.return_value = 1
+
+    jwt_manager.session_ttl = 5
+    payload = jwt_manager._jwt_payload_from_user_profile(profile)
+
+    assert payload == {
+        'urs-user-id': 'test_user_id',
+        'first_name': 'test_first_name',
+        'last_name': 'test_last_name',
+        'email': 'test_email',
+        'urs-access-token': 'test_token',
+        'urs-groups': ['test_group1', 'test_group2'],
+        'iat': 1,
+        'exp': 6
+    }
+
+
+@mock.patch(f'{MODULE}.JwtManager._in_blacklist')
```

Comment:
You should use `autospec=True` as much as possible just to make sure that the mocked functions are still called with the correct number/names of arguments.

```suggestion
@mock.patch(f'{MODULE}.JwtManager._in_blacklist', autospec=True)
```